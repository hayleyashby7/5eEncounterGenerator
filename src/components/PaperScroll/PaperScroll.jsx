import React from 'react';
import PropTypes from 'prop-types';

function PaperScroll({ contents }) {
    return (
        <div className='relative flex-col bg-orange-100 bg-[size:150%] text-red-950 before:absolute before:bottom-full before:left-0 before:h-5 before:w-full before:rotate-180 before:bg-paper before:bg-bottom after:absolute after:left-0 after:top-full after:h-5 after:w-full after:bg-paper after:bg-bottom'>
            {contents && contents}
        </div>
    );
}

export default PaperScroll;

PaperScroll.propTypes = {
    contents: PropTypes.node.isRequired,
};
