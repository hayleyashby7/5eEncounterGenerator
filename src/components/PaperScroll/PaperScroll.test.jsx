import { describe, it, expect } from 'vitest';
import React from 'react';
import { render, screen } from '@testing-library/react';
import PaperScroll from './PaperScroll';

describe('PaperScroll component', () => {
    it('renders component', () => {
        // Arrange
        const contents = <div>Test</div>;

        // Act
        const { container } = render(<PaperScroll contents={contents} />);

        // Assert
        expect(container).toMatchSnapshot();
    });

    it('renders component with contents', () => {
        // Arrange
        const contents = <div>Test</div>;

        // Act
        render(<PaperScroll contents={contents} />);

        // Assert
        expect(screen.getByText(/Test/i)).toBeInTheDocument();
    });
});
