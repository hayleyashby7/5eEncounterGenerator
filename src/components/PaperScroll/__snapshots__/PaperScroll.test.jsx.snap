// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PaperScroll component > renders component 1`] = `
<div>
  <div
    class="relative flex-col bg-orange-100 bg-[size:150%] text-red-950"
  >
    <span
      class="absolute bottom-full left-0 h-5 w-full rotate-180 bg-paper bg-bottom"
    />
    <div>
      Test
    </div>
    <span
      class=" absolute left-0 top-full h-5 w-full bg-paper bg-bottom"
    />
  </div>
</div>
`;
